apps/nextjs/src/components/ContextProviders/ChatProvider.tsx

```diff
@@ -125,8 +126,7 @@ export function ChatProvider({
   const chatAreaRef = useRef<HTMLDivElement>(null);
   const [hasFinished, setHasFinished] = useState(true);
 
-  const [hasAppendedInitialMessage, setHasAppendedInitialMessage] =
-    useState(false);
+  const hasAppendedInitialMessage = useRef<boolean>(false);
```

Comment:
Changed this as `append` was being called multiple times on the initial message, resulting in bizarre behaviour with ids

Reply:
Interesting. So in a sense, this change to a useRef is just to satisfy the linter error when you remove it from the useEffect's watchlist?